### PR TITLE
[hold][WIP] Feature: Download data in CSV format [LEI-246]

### PR DESCRIPTION
**Pull request by samanehsan**
Refs: https://openscience.atlassian.net/browse/LEI-246

## Purpose
- Add download as CSV option to experimenter responses page
- Add download as CSV (for ISP) option that 

## Summary of changes
- Change TSV converter to CSV 
- Format ISP field names so that they do not include the frameId, just the field titles from the code book

## Testing notes
- Test that any fields containing quotes, newline characters, commas, etc. are properly formatted
- Test that the field names correctly correspond to the data

### CODE REVIEW

File: app/components/export-tool.js [hunk 1]
```diff
@@ -82,44 +82,54 @@ export default Ember.Component.extend({
...
 
         var fields = Object.keys(squashed[0]);
-        var tsv = [fields.join('\t')];
+        var csv = [fields.join(',')];
 
         squashed.forEach((item) => {
             var line = [];
             fields.forEach(function(field) {
-                line.push(JSON.stringify(item[field]));
+                var value = JSON.stringify(item[field]).replace(/\\"/g, '""');
```
Comment (abought) [inline app/components/export-tool.js]: There may be additional edge cases to consider. For example, what if the field contains a comma?

This document may be helpful for you (see section 2 in particular):
https://tools.ietf.org/html/rfc4180#section-2

File: app/components/export-tool.js [hunk 2]
```diff
@@ -82,44 +82,54 @@ export default Ember.Component.extend({
...
         return JSON.stringify(dataArray, null, 4);
     },
-    _convertToTSV(dataArray) {
+    _convertToCSV(dataArray) {
         // Flatten the dictionary keys for readable column headers
         let squashed = dataArray.map((item => squash(item)));
 
         var fields = Object.keys(squashed[0]);
-        var tsv = [fields.join('\t')];
+        var csv = [fields.join(',')];
```
Comment (abought) [inline app/components/export-tool.js:90]: Objects are a hash, and records may not all have the same data. Make sure that the correct fields from each record line up from row to row. 

(don't just assume that the second field is always known)

Eg, how python does it:
https://docs.python.org/2/library/csv.html#csv.DictWriter

Getting the list of field names across all records will also be a challenge to address.

**Comment by abought** [comment]
Other testing notes: since CSV isn't a 100% consistent standard, the ultimate test is that the file can be opened by the client. It's not a bad idea to spend 5-10 min very quickly verifying that it opens in common programs like LibreOffice, Excel, or Python/R.

(I have LibreOffice on my work machine and excel at home if it helps)

**Comment by samanehsan** [comment]
The download as CSV option includes all session data, including some fields that are lists (e.g. `sequence`. The resulting CSV file interprets the list as multiple cells instead of just one cell. These should be removed where possible and the CSV formatter should turn any lists/objects that make it through into something that won't break the CSV file (perhaps one long string?)

**Comment by abought** [comment]
This work has been incorporated into another PR here:
https://github.com/CenterForOpenScience/exp-addons/pull/205

(along with a planned followup that @binoculars is working on to integrate it into experimenter for the ISP-specific use case)

Thanks for getting things moving on this one!

**Pull request closed without merging**